File: Core/Frontend/Timer.py
import time

class Timer(object):

    def __init__(self):
        self.__mBegin = None
        self.__mEnd = None

    def StartClockNow(self):
        self.__mBegin = time.time()

    def StopClockNow(self):
        self.__mEnd = time.time()

    def TimeElapsedMilliSec(self):
        return round((self.__mEnd - self.__mBegin) * 1000)

File: Core/Backend/Timer.cc
#include "Timer.h"

namespace TinySpark
{
    Timer::Timer() {}
    Timer::~Timer() {}
    void Timer::StartClockNow() { m_Begin = std::chrono::high_resolution_clock::now(); }
    void Timer::StopClockNow() { m_End = std::chrono::high_resolution_clock::now(); }
    float Timer::TimeElapsedNanoSec() { return (m_End - m_Begin).count(); }
    float Timer::TimeElapsedMilliSec() { return (m_End - m_Begin).count() * 0.000001f; }
    // float Timer::TimeElapsedSec();
} // namespace TinySpark